PhysicsTools/NanoAOD/plugins/TriggerObjectTableProducer.cc

```diff
@@ -91,8 +91,37 @@ class TriggerObjectTableProducer : public edm::stream::EDProducer<> {
           l1DR2_2(-1),
           l2DR2(-1),
           skipObjectsNotPassingQualityBits(pset.getParameter<bool>("skipObjectsNotPassingQualityBits")),
-          qualityBits(pset.getParameter<std::string>("qualityBits")),
-          qualityBitsDoc(pset.getParameter<std::string>("qualityBitsDoc")) {
+          qualityBits("0"),   //will be overwritten from configuration
+          qualityBitsDoc("")  //will be created from configuration
+    {
+      if (pset.existsAs<std::string>("qualityBits")) {
+        qualityBits = StringObjectFunction<pat::TriggerObjectStandAlone>(pset.getParameter<std::string>("qualityBits"));
+        qualityBitsDoc = pset.getParameter<std::string>("qualityBitsDoc");
+      } else {
+        std::vector<edm::ParameterSet> qualityBitsConfig =
+            pset.getParameter<std::vector<edm::ParameterSet>>("qualityBits");
+        std::stringstream qualityBitsFunc;
+        std::vector<bool> bits(qualityBitsConfig.size(), false);
+        for (size_t i = 0; i != qualityBitsConfig.size(); ++i) {
```

Comment:
Should we check that `i` doesn't get too big? (32 in practice, with the option to switch to int64 if at some point we need to add more filters)